pandas/tseries/resample.py

```diff
@@ -14,7 +21,560 @@
+                                               **kwargs)
+
+        return result
+
+    agg = aggregate
+    apply = aggregate
+
+    def transform(self, arg, *args, **kwargs):
+        """
+        Provide a transformation on the Resampler
+        Return the same size as input
+        """
+        return self._selected_obj.groupby(self.groupby).transform(
+            arg, *args, **kwargs)
+
+    def _downsample(self, f):
+        raise AbstractMethodError(self)
+
+    def _upsample(self, f, limit=None):
+        raise AbstractMethodError(self)
+
+    def _gotitem(self, key, ndim, subset=None):
+        """
+        sub-classes to define
+        return a sliced object
+
+        Parameters
+        ----------
+        key : string / list of selections
+        ndim : 1,2
+            requested ndim of result
+        subset : object, default None
+            subset to act on
+        """
+        self._set_binner()
+        grouper = self.grouper
+        if subset is None:
+            subset = self.obj
+        grouped = groupby(subset, by=None, grouper=grouper, axis=self.axis)
+
+        # try the key selection
+        try:
+            return grouped[key]
+        except KeyError:
+            return grouped
+
+    def _groupby_and_aggregate(self, grouper, how, *args, **kwargs):
+        """ revaluate the obj with a groupby aggregation """
+
+        if grouper is None:
+            self._set_binner()
+            grouper = self.grouper
+
+        obj = self._selected_obj
+
+        try:
+            grouped = groupby(obj, by=None, grouper=grouper, axis=self.axis)
+        except TypeError:
+
+            # panel grouper
+            grouped = PanelGroupBy(obj, grouper=grouper, axis=self.axis)
+
+        try:
+            result = grouped.aggregate(how, *args, **kwargs)
+        except Exception:
+
+            # we have a non-reducing function
+            # try to evaluate
+            result = grouped.apply(how, *args, **kwargs)
+
+        return self._wrap_result(result)
+
+    def _wrap_result(self, result):
+        """ potentially wrap any results """
+        return result
+
+    def pad(self, limit=None):
+        """
+        Forward fill the values
+
+        Parameters
+        ----------
+        limit : integer, optional
+            limit of how many values to fill
+        """
+        return self._upsample('pad', limit=limit)
+    ffill = pad
+
+    def backfill(self, limit=None):
+        """
+        Backward fill the values
+
+        Parameters
+        ----------
+        limit : integer, optional
+            limit of how many values to fill
+        """
+        return self._upsample('backfill', limit=limit)
+    bfill = backfill
+
+    def fillna(self, method, limit=None):
+        """
+        Parameters
+        ----------
+        limit : integer, optional
+            limit of how many values to fill
+        """
+        return self._upsample(None, limit=limit)
```

Comment:
In the test (https://github.com/jreback/pandas/blob/78c909ceb780c2bdb9d8df878089a23032a19e0f/pandas/tseries/tests/test_resample.py#L230, can't comment on it as the diff is too big, a bit annoying), you are downsampling instead of upsampling, that is the reason I think the test does not catch this